src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -20,75 +18,63 @@ namespace Stratis.Bitcoin.P2P
     /// </summary>
     public sealed class PeerAddressManager : IDisposable
     {
-        private IAsyncLoop asyncLoop;
-        private readonly IAsyncLoopFactory asyncLoopFactory;
-        private readonly string peerFilePath;
-        internal const string PEERFILENAME = "peers.json";
+        private DataFolder peerFilePath;
+        internal const string PeerFileName = "peers.json";
 
-        public PeerAddressManager(IAsyncLoopFactory asyncLoopFactory, string peerFilePath)
+        public PeerAddressManager(DataFolder peerFilePath)
         {
-            Guard.NotNull(asyncLoopFactory, nameof(asyncLoopFactory));
-
-            this.asyncLoopFactory = asyncLoopFactory;
             this.peerFilePath = peerFilePath;
+            this.Peers = new ConcurrentDictionary<IPEndPoint, PeerAddress>();
         }
 
-        private ConcurrentDictionary<int, PeerAddress> peers = new ConcurrentDictionary<int, PeerAddress>();
-        public ConcurrentDictionary<int, PeerAddress> Peers
-        {
-            get { return this.peers; }
-        }
+        /// <summary>
+        /// A key value store that indexes all discovered peers by their end point.
+        /// </summary>
+        public ConcurrentDictionary<IPEndPoint, PeerAddress> Peers { get; private set; }
 
         public void LoadPeers()
         {
             var fileStorage = new FileStorage<List<PeerAddress>>(this.peerFilePath);
-            var peers = fileStorage.LoadByFileName(PEERFILENAME);
+            var peers = fileStorage.WithConverters(new[] { new IPEndpointConverter() }).LoadByFileName(PeerFileName);
             peers.ForEach(peer =>
             {
-                this.peers.TryAdd(peer.NetworkAddress.Endpoint.GetHashCode(), peer);
+                this.Peers.TryAdd(peer.NetworkAddress.Endpoint, peer);
             });
         }
 
         public void SavePeers()
         {
-            if (this.peers.Any() == false)
+            if (this.Peers.Any() == false)
                 return;
 
             var fileStorage = new FileStorage<List<PeerAddress>>(this.peerFilePath);
-            fileStorage.SaveToFile(this.peers.Select(p => p.Value).ToList(), PEERFILENAME);
+            fileStorage.WithConverters(new[] { new IPEndpointConverter() }).SaveToFile(this.Peers.Select(p => p.Value).ToList(), PeerFileName);
         }
 
         public void AddPeer(NetworkAddress networkAddress, IPAddress source)
         {
-            //[NBitcoin] what is this check?
+            //Specific IP address ranges that are reserved specifically as non - routable addresses to be used in 
```

Comment:
Ah I mean this comment can go in the `IsRoutable` method

Reply:
Ahhh! Ok cool, I moved it there 👍